askama_derive/src/generator.rs

```diff
@@ -727,6 +727,57 @@ impl<'a> Generator<'a> {
                 args.len()
             )));
         }
+        // Handling both named and unnamed arguments requires to be careful of the arguments order.
+        // To do so, we iterate through the macro defined arguments and then check if we have a
+        // named argument with this name:
+        //
+        // * If there is one, we add it and move to the next argument.
+        // * If there isn't one, we then need to pick the next unnamed argument (as we consider they
+        //   are passed in the right order). To ensure we won't accidentally pass a named argument,
+        //   we split named and unnamed arguments in two vectors before starting.
+        let mut output_args = Vec::new();
+        // Since named arguments can only be passed last, we only need to check if the last argument
+        // is a named one.
+        let args = if args
```

Comment:
I think this can be simplified to `if let Some(Expr::NamedArgument(_, _)) = args.last()`?

Reply:
Much better indeed!